test/getElementsByTagName.spec.js

```diff
@@ -0,0 +1,312 @@
+        },
+        {
+          tagName: 'DIV',
+          classList: ['container']
+        },
+        {
+          tagName: 'DIV',
+          classList: ['outer-component']
+        },
+        {
+          tagName: 'DIV',
+          classList: ['wrapper']
+        },
+        {
+          tagName: 'DIV',
+          classList: ['inner-component']
+        },
+        {
+          tagName: 'SPAN',
+          classList: []
+        },
+        {
+          tagName: 'svg',
+          classList: []
+        },
+        {
+          tagName: 'defs',
+          classList: []
+        },
+        {
+          tagName: 'linearGradient',
+          classList: []
+        },
+        {
+          tagName: 'stop',
+          classList: []
+        },
+        {
+          tagName: 'stop',
+          classList: []
+        },
+        {
+          tagName: 'circle',
+          classList: []
+        }
+    ];
+
+    const expectedWildcardShadow = [
+        {
+          tagName: 'HEAD',
+          classList: []
+        },
+        {
+          tagName: 'BODY',
+          classList: []
+        },
+        {
+          tagName: 'DIV',
+          classList: ['container']
+        },
+        {
+          tagName: 'OUTER-COMPONENT',
+          classList: []
+        },
+        {
+          tagName: 'DIV',
+          classList: ['wrapper']
+        },
+        {
+          tagName: 'INNER-COMPONENT',
+          classList: []
+        },
+        {
+          tagName: 'SPAN',
+          classList: []
+        },
+        {
+          tagName: 'svg',
+          classList: []
+        },
+        {
+          tagName: 'defs',
+          classList: []
+        },
+        {
+          tagName: 'linearGradient',
+          classList: []
+        },
+        {
+          tagName: 'stop',
+          classList: []
+        },
+        {
+          tagName: 'stop',
+          classList: []
+        },
+        {
+          tagName: 'circle',
+          classList: []
+        },
+        {
+          tagName: 'SCRIPT',
+          classList: []
+        }
+    ];
+    testTagNames('handles wildcard tag name', '*', tagNameLight1, tagNameShadow1, expectedWildcard, expectedWildcardShadow);
+    
+    const expectedCaseInsensitive = [
+        {
+            tagName: 'SPAN',
+            classList: []
+        }
+    ];
+    testTagNames('is case insensitive', 'SpAn', tagNameLight1, tagNameShadow1, expectedCaseInsensitive);
+  })
+
+  describe('getElementsByTagNameNS', () => {
+    testTagNamesNS('returns empty with falsy ns', '', '*', tagNameLight1, tagNameShadow1, []);
+    testTagNamesNS('returns empty with falsy ns', false, '*', tagNameLight1, tagNameShadow1, []);
+    testTagNamesNS('returns empty with falsy ns', undefined, '*', tagNameLight1, tagNameShadow1, []);
+    testTagNamesNS('returns empty with falsy ns', null, '*', tagNameLight1, tagNameShadow1, []);
+    testTagNamesNS('returns empty with falsy ns', 0, '*', tagNameLight1, tagNameShadow1, []);
```

Comment:
Great tests! If we had actual code coverage tests, I'm sure they would show you're getting great coverage here. 🙂

Reply:
I'll create an issue for adding test coverage.